src/main/scala/tech/cryptonomic/conseil/util/DatabaseUtil.scala

```diff
@@ -145,15 +151,23 @@ object DatabaseUtil {
       *
       * @param table   table on which query will be executed
       * @param columns columns which are selected from teh table
-      * @param aggregation parameter containing info about field which has to be aggregated
+      * @param aggregations parameter containing info about field which has to be aggregated
       * @return SQLAction with basic query
       */
-    def makeQuery(table: String, columns: List[String], aggregation: List[Aggregation]): SQLActionBuilder = {
-      val aggregationFields = aggregation
-        .map { aggr =>
-          mapAggregationToSQL(aggr.function, aggr.field) + " as " + mapAggregationToAlias(aggr.function, aggr.field)
-        }
-      val aggr = aggregationFields ::: columns.toSet.diff(aggregation.map(_.field).toSet).toList
+    def makeQuery(table: String, columns: List[Field], aggregations: List[Aggregation]): SQLActionBuilder = {
+      val aggregationFields = aggregations.map {
+        aggregation =>
+          mapAggregationToSQL(aggregation.function, aggregation.field) + " as " + mapAggregationToAlias(
+            aggregation.function,
+            aggregation.field
+          )
+      }
+      val ccc = columns.map {
+        case SimpleField(field) => field
+        case FormattedField(field, function, format) =>
+          makeAggregationFormat(field, format) + " as " + mapFormatToAlias(function, field)
+      }
+      val aggr = aggregationFields ::: ccc.toSet.diff(aggregations.map(_.field).toSet).toList
```

Comment:
Not quite clear to me what's happening here, would some code comment and example clear up the logic?